lib/matplotlib/animation.py

```diff
@@ -1156,8 +1160,9 @@ def _init_draw(self):
             self._draw_frame(next(self.new_frame_seq()))
         else:
             self._drawn_artists = self._init_func()
-            for a in self._drawn_artists:
-                a.set_animated(self._blit)
+            if iterable(self._drawn_artists):
```

Comment:
Right. My point is that we are letting things go through without anything
being done. Silent inaction is hard to debug. Although, I would be fine
with allowing a single artist to be returned as well, as it is trivial to
support as well.
On Aug 27, 2015 10:45 PM, "Thomas A Caswell" <notifications@github.com>
wrote:

> In lib/matplotlib/animation.py
> <https://github.com/matplotlib/matplotlib/pull/4995#discussion_r38167498>:
>
> > @@ -1156,8 +1160,9 @@ def _init_draw(self):
> >              self._draw_frame(next(self.new_frame_seq()))
> >          else:
> >              self._drawn_artists = self._init_func()
> > -            for a in self._drawn_artists:
> > -                a.set_animated(self._blit)
> > +            if iterable(self._drawn_artists):
>
> On the other hand, my understanding of the documentation is that the user
> function are suppossed to return a list of artists.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4995/files#r38167498>.
>